ci_scripts/assert_news.py

```diff
@@ -1,86 +1,153 @@
-
-    # the output of the git command looks like:
-    # '  HEAD branch: master'
-    origin_branch = 'master'  # unless we prove otherwise
-    for line in origin_stats.splitlines():
-        if 'head branch:' in line.lower():
-            origin_branch = line.split(':', 1)[-1].strip()
-            break
-
-    origin = '%s/%s' % (remote_alias, origin_branch)
-
-    # figure out the current branch
-    current_branch = subprocess.check_output(
-        ['git', 'rev-parse', '--abbrev-ref', 'HEAD'],
-        cwd=news_dir
-    ).decode().strip()
-
-    print(':: Finding news in `%s` to add to remote `%s`' % (current_branch, origin))
-    diff_command = ['git', 'diff', '%s...%s' % (origin, current_branch), '--name-status', news_dir]
-    file_diff = subprocess.check_output(
-        diff_command,
-        cwd=news_dir
-    ).decode()
-
-    # the output of the git command looks like:
-    # 'A       docs/news/789.feature'
-
-    # [optional] ensure we have an addition, rather than just modify/delete
-    added_news = [line for line in file_diff.splitlines() if line.lower().strip().startswith('a')]
-
-    # pass or fail
-    if not added_news:
-        print(
-            '! Error: Uh-oh, did not find any news files!\n'
-            '! Please add a news file to `%s`\n'
-            '+ File diff:\n%s\n'
-            '{} Git diff command:\n`%s`\n' % (
-                news_dir,
-                file_diff.strip(),
-                subprocess.list2cmdline(diff_command)
-            )
-        )
-        exit(1)  # exit with an error status, no need for a traceback
-    print(':: %s new files in `%s`' % (len(added_news), news_dir))
-
-
-if __name__ == '__main__':
-    news_dir = sys.argv[1] if len(sys.argv) > 1 else None
-    main(news_dir)
+import logging
+import os
+import re
+import sys
+import argparse
+from utils.definitions import PROJECT_ROOT, MASTER_BRANCH, BETA_BRANCH, \
+    RELEASE_BRANCH_PATTERN, NEWS_DIR
+from utils.git_helpers import GitWrapper
+from utils.logging import log_exception, set_log_level
+
+logger = logging.getLogger(__name__)
+
+VALID_EXTENSIONS = ['misc', 'doc', 'removal', 'bugfix', 'feature', 'major']
+NEWS_FILE_NAME_FORMAT = r"^[0-9]+.[a-z]+$"
+
+"""
+Ensures news file are created for all new changes to the project.
+"""
+
+
+class NewsFileValidator(object):
+    """
+    Verification of the individual news files: naming, existence, content
+    """
+
+    def __init__(self, full_path):
+        self.news_file_path = full_path
+
+    def _verify_news_file_name(self):
+        basename = str(os.path.basename(
+            self.news_file_path)) if self.news_file_path else None
+        if (not basename) or (not re.search(NEWS_FILE_NAME_FORMAT, basename)):
+            raise ValueError(
+                f'Incorrect news file name [{basename}]. It must follow the following format: {NEWS_FILE_NAME_FORMAT}'
+            )
+        for extension in VALID_EXTENSIONS:
+            if extension in basename:
+                return

```

Comment:
If a break then else is used then then both validation rules follow the same flow and it also allows extra validation rules to be added in future following the same pattern.

```suggestion
                break
        else:
```